probe/probe.go

```diff
@@ -210,11 +210,12 @@ func (p *Probe) tag(r report.Report) report.Report {
 
 func (p *Probe) drainAndPublish(rpt report.Report, rs chan report.Report) {
 	p.rateLimiter.Wait(context.Background())
+	rpt = rpt.Copy()
```

Comment:
Nit: Does the `rpt report.Report` in the function arg only do a shallow copy? If it's a deep copy, then we don't need this line, if it's shallow, then why not only pass the reference there `rpt *report.Report`?

Reply:
Since Report holds some slices and maps, copying it is shallow so we need the deep copy.
However in Go it is idiomatic to pass a reference only where you intend to modify the data, or when profiling shows the reference to be way more efficient.